src/validation.cpp

```diff
@@ -2565,7 +2565,11 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
             uiInterface.NotifyBlockTip(fInitialDownload, pindexNewTip);
         }
     } while (pindexNewTip != pindexMostWork);
-    CheckBlockIndex(chainparams.GetConsensus(pindexNewTip->nHeight));
+    if (pindexNewTip != NULL) {
+        CheckBlockIndex(chainparams.GetConsensus(pindexNewTip->nHeight));
+    } else {
+        CheckBlockIndex(chainparams.GetConsensus(0));
```

Comment:
Errr... isn't this correct in any scenario? As in, if we have no active tip, then we're at genesis anyway?

Reply:
Hmm yes, you are correct. However in reading the code that loops right before this, I now wonder which test caused this issue - I'm taking this under investigation because we may be obscuring a bug this way?

Reply:
Don't think we have the previous CI runs to check it against, if I remember correctly it was after all the tests in `bumpfee.py` run. You should be able to re-create by reverting the crash fix and then running `bumpfee.py` a few times, and look for an exit code of `-11`.

Meanwhile thoughts on getting the code improvements in? While the crash doesn't appear likely in production it would be good to get that fix in certainly, and generally I'd rather start enabling tests.

Reply:
This PR was meant to change 1 wallet parameter to fix an error of judgement in the wallet configuration made 4 years ago. Subsequently though, this added:

1. A missing implementation in `rpcwallet.cpp`
2. A test that never passed before (because of the missing implementation under 1 AND because the test was never ported)
3. A change to `validation.cpp` to solve an unexplained crash. (no root cause identified)

Additionally, [please note that RBF is not even enabled as a default policy in Dogecoin Core](https://github.com/dogecoin/dogecoin/blob/f80bfe9068ac1a0619d48dad0d268894d926941e/src/wallet/wallet.h#L68) so the impact of the change you're pushing is extremely low.

Because the test fixes are building **_on top_** of a policy change rather than before we make a policy change, like we are doing with all other planned policy changes, I need to know root causes to problems we're solving and am going to have to rip this apart. Sorry 😢

Reply:
> As in, if we have no active tip, then we're at genesis (height 0) anyway?

This unfortunately is not the case for the bumpfee test where this occurred, the node already had 71 blocks in the active tip. I inserted a log message `ActivateBestChain: No best chain found while processing block %s` before line 2571 in your else block:

```
2021-08-03 18:53:54 UpdateTip: new best=799d6e989dbb31b27ba994312fa092edcc806646018bccb945536d03e8781f1e height=71 version=0x00620004 log2_work=7.169925 tx=97 date='2021-08-03 18:54:06' progress=1.000000 cache=0.0MiB(96tx)

[..]

2021-08-03 18:54:00 received: inv (37 bytes) peer=0
2021-08-03 18:54:00 got inv: block ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc  new peer=0
2021-08-03 18:54:00 sending getheaders (581 bytes) peer=0
2021-08-03 18:54:00 getheaders (71) ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc to peer=0
2021-08-03 18:54:00 received: inv (37 bytes) peer=1
2021-08-03 18:54:00 got inv: block ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc  new peer=1
2021-08-03 18:54:00 sending getheaders (581 bytes) peer=1
2021-08-03 18:54:00 getheaders (71) ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc to peer=1
2021-08-03 18:54:00 received: headers (82 bytes) peer=0
2021-08-03 18:54:00 Requesting block ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc from  peer=0
2021-08-03 18:54:00 sending getdata (37 bytes) peer=0
2021-08-03 18:54:00 received: cmpctblock (309 bytes) peer=0
2021-08-03 18:54:00 Initialized PartiallyDownloadedBlock for block ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc using a cmpctblock of size 309
2021-08-03 18:54:00 sending getblocktxn (41 bytes) peer=0
2021-08-03 18:54:00 received: headers (82 bytes) peer=1
2021-08-03 18:54:00 received: blocktxn (1844 bytes) peer=0
2021-08-03 18:54:00 Successfully reconstructed block ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc with 1 txn prefilled, 4 txn from mempool (incl at least 0 from extra pool) and 8 txn requested
2021-08-03 18:54:00 Calling ActivateBestChain() from ProcessNewBlock (validation.cpp)
2021-08-03 18:54:00 ActivateBestChain: No best chain found while processing block ab0c0563e8e5d98299c7c7c56e2a8aa71418b1f818f7f4dd9709b8a8a13792fc
```

[full debug.log from the node where the error occurs](https://github.com/dogecoin/dogecoin/files/6926509/debug.txt) -  can reproduce this consistently.

Reply:
Fixed the root cause in #2434. Merging that resolves this comment thread.